### PR TITLE
MCU board support for Raspberry Pi Pico 2

### PR DESCRIPTION
**Pull request by burbokop**
ChangeLog: MCU board support for Raspberry Pi Pico 2
This pull request is not related to any issue. 

Tested on real hardware.

There are two files copied from other projects:
```
examples/mcu-board-support/pico2_st7789/memory.x
examples/mcu-board-support/pico2_st7789/rp_pico2.rs
```
Inside you can find comments where it was copied from. How should i proceed to do everything legally right?

### CODE REVIEW

File: examples/mcu-board-support/README.md [hunk 1]
```diff
@@ -74,6 +74,28 @@ udisksctl mount -b /dev/sda1
...
 
+### On the Raspberry Pi Pico2
+
+Build the demo with:
+
+```sh
+cargo build -p printerdemo_mcu --no-default-features --features=mcu-board-support/pico2-st7789 --target=thumbv8m.main-none-eabihf --release
+```
+
+The resulting file can be flashed conveniently with [picotool](https://github.com/raspberrypi/picotool). You shold build it from source:
```
Comment (tronical) [inline examples/mcu-board-support/README.md:85]: Or did you mean to provide instructions for picotool here? (I removed the `:`)

```suggestion
The resulting file can be flashed conveniently with [picotool](https://github.com/raspberrypi/picotool). You should build it from source.
```

File: examples/mcu-board-support/README.md [hunk 2]
```diff
@@ -74,6 +74,28 @@ udisksctl mount -b /dev/sda1
...
+
+Then upload the demo to the Raspberry Pi Pico: push the "bootsel" white button on the device while connecting the
+micro-usb cable to the device, this connect some storage where you can store the binary.
```
Comment (tronical) [inline examples/mcu-board-support/README.md:88]:
```suggestion
micro-usb cable to the device, this connects some USB storage on your workstation where you can store the binary.
```

File: examples/mcu-board-support/README.md [hunk 3]
```diff
@@ -74,6 +74,28 @@ udisksctl mount -b /dev/sda1
...
+
+Or from the command on linux: (connect the device while pressing the "bootsel" button.
```
Comment (tronical) [inline examples/mcu-board-support/README.md:90]:
```suggestion
Or from the command on linux (connect the device while pressing the "bootsel" button):
```

File: examples/mcu-board-support/pico2_st7789/memory.x [hunk 4]
```diff
@@ -0,0 +1,79 @@
+/*
+Copied from https://github.com/rp-rs/rp-hal/blob/main/rp235x-hal-examples/memory.x
+*/
```
Comment (tronical) [inline examples/mcu-board-support/pico2_st7789/memory.x:3]:
```suggestion
/* Copyright © 2021 rp-rs organization
 SPDX-License-Identifier: MIT OR Apache-2.0
 Copied from https://github.com/rp-rs/rp-hal/blob/main/rp235x-hal-examples/memory.x
*/
```

File: examples/mcu-board-support/pico2_st7789/rp_pico2.rs [hunk 5]
```diff
@@ -0,0 +1,820 @@
+/*
+Copied from https://github.com/rp-rs/rp-hal-boards/blob/main/boards/rp-pico/src/lib.rs
+and slightly changed for RP2350 chip
+*/
```
Comment (tronical) [inline examples/mcu-board-support/pico2_st7789/rp_pico2.rs:4]:
```suggestion
/* Copyright © 2021 rp-rs organization
 SPDX-License-Identifier: MIT OR Apache-2.0
 Copied from https://github.com/rp-rs/rp-hal-boards/blob/main/boards/rp-pico/src/lib.rs
 and slightly changed for RP2350 chip
*/
```

**Review by tronical — COMMENTED** [review]
Thanks for your contribution!

AFAICS the license from rp-rs are licensed under MIT/Apache, so I've included suggested corrections.

I've also made a few other small suggestions.

The last issue that I see left (before the CI discovers any new... ;-) is that `pico2_st7789.rs` needs reformatting.

**Comment by burbokop** [comment]
It seems docs checker is not happy with "Copyright © 2021 rp-rs organization"

**Comment by tronical** [comment]
Indeed. I've pushed a fix into your branch. Fingers crossed :)

**Review by tronical — APPROVED** [review]

**Comment by tronical** [comment]
Thanks @burbokop!

**Pull request merged**